src/transformers/integrations.py

```diff
@@ -230,6 +233,149 @@ def rewrite_logs(d):
     return new_d
 
 
+def init_deepspeed(trainer, num_training_steps):
+    """
+    Init DeepSpeed, after converting any relevant Trainer's args into DeepSpeed configuration
+
+    Args:
+        trainer: Trainer object
+        num_training_steps: per single gpu
+
+    Returns: model, optimizer, lr_scheduler
+    """
+    import io
+    import json
+    from types import SimpleNamespace
```

Comment:
Those can go at the beginning of the file however since there are standard lib (fast) imports.